File: bin/main/static/js/service/CustomerService.js
app.factory('CustomerService', CustomerService);

function CustomerService($resource) {
	/*var service = {};
	service.getAllCustomer = getAllCustomer;
	service.getUserById = getUserById;
	
	return service;
	
	function getAllCustomer() {
		return $http.get("/customer/all");
	}
	
	function getUserById(id) {
		return $http.get("/customer/" + id);
	}*/
	return $resource('/customer/:id');
}

File: src/main/java/com/lbn/service/CustomerService.java
package com.lbn.service;

import com.lbn.domain.Customer;

public interface CustomerService {
	public Customer findById(int id);
	public Customer findByPhone(String phone);
	public Customer findByName(String name);
	public Customer findByCompany(String company);
	public boolean saveCustomer(Customer customer);
	public void deleteById(int id);
	public Iterable<Customer> getAllCustomer();
}
